Rakefile

```diff
@@ -425,3 +425,288 @@ end
 
 spec = Gem::Specification.find_by_name 'gettext-setup'
 load "#{spec.gem_dir}/lib/tasks/gettext.rake"
+
+namespace :pwsh do
+  desc "Generate pwsh from Bolt's command line options"
+  task :generate_module do
+    @pwsh_verbs = {
+      'apply'      => 'Invoke',
+      'convert'    => 'Convert',
+      'createkeys' => 'New',
+      'init'       => 'New',
+      'install'    => 'Install',
+      'encrypt'    => 'Protect',
+      'decrypt'    => 'Unprotect',
+      'migrate'    => 'Update',
+      'run'        => 'Invoke',
+      'show'       => 'Get',
+      'upload'     => 'Send' # deploy? publish?
+    }
+
+    @hardcoded_cmdlets = {
+      'createkeys' => {
+        'verb' => 'New',
+        'noun' => 'BoltSecretKey'
+      },
+      'show-modules' => {
+        'verb' => 'Get',
+        'noun' => 'BoltPuppetfileModules'
+      },
+      'generate-types' => {
+        'verb' => 'Register',
+        'noun' => 'BoltPuppetfileTypes'
+      }
+    }
+
+    parser = Bolt::BoltOptionParser.new({})
+
+    @commands = []
+    Bolt::CLI::COMMANDS.each do |subcommand, actions|
+      actions << nil if actions.empty?
+      actions.each do |action|
+        help_text = parser.get_help_text(subcommand, action)
+        matches = help_text[:banner].match(/USAGE(?<usage>.+?)DESCRIPTION(?<desc>.+?)(EXAMPLES|\z)/m)
+        action.chomp unless action.nil?
+
+        if action.nil? && subcommand == 'apply'
```

Comment:
It seems like this is covered by the `else` clause in this statement?

Reply:
The `else` clause didn't cover `bolt apply` which is different than all the other commands because it has no action. I couldn't find an if/else or case statement that worked and looked nice, but I found if I made `apply` an action the original if statement works while keeping the overall intent *and* making sure it fails if something new is added we don't have a mapping for.

Reply:
I have left it with the if/elseif/else/end block because I need to account for `bolt apply` being different from all other commands in that it has no action. There is probably a more clever way of expressing it, but I think the code is readable enough for future use and it works